### ISSUE TITLE
FormatException: Unexpected character

### ISSUE THREAD

**Issue by mutairibassam** [AymanProjects/dart-flavorz#10]
Format exception error will be thrown in case there is a `,` comma in the last value of an object. For simplicity I added an example below for a single object.

Below example will fail:
```json
{
    "environmentToRun": "mock",
    "environments": [
        {
            "_name": "mock",
            "versionNumber": "mock 1.0.0",
        }
    ]
}
```

Below example will generate successfully:
```json
{
    "environmentToRun": "mock",
    "environments": [
        {
            "_name": "mock",
            "versionNumber": "mock 1.0.0"
        }
    ]
}
```

The difference is only a comma in `local` object, such cases should be handled by the package.

**Comment by AymanProjects** [comment]
@mutairibassam as far as I know, the first sample is not a valid json format. 
The json file should be properly formatted.

**Comment by mutairibassam** [comment]
I agree. These kinds of silly things should be handled by package/application level to provide a better user experience. For instance, if you tried to validate the improper json through this [website ](https://jsonformatter.curiousconcept.com/#) it will validate and correct. On the other hand if you tried to validate using this [website](https://jsonlint.com/) it will show improper format. The first website is way better because it's validate and correct.

Not all users are equal, so sometimes we need to do background work just to increase packages' popularity.

**Comment by AymanProjects** [comment]
`Flavorz` package uses a built-in json decoder tool that converts json into string. 
The tool itself doesn't take care of the trailing comma. 

We can't build a new decoding tool just to solve the above issue. However, I will leave this issue open for people who wants to contribute. 

Thank you @mutairibassam
